src/patternfly/components/ModalBox/styles.scss

```diff
@@ -2,85 +2,186 @@
+  max-width: calc(100% - var(--pf-c-modal-box--MarginLeft) - var(--pf-c-modal-box--MarginRight));
   min-height: var(--pf-c-modal-box--MinHeight);
   max-height: var(--pf-c-modal-box--MaxHeight);
   background-color: var(--pf-c-modal-box--BackgroundColor);
   border: var(--pf-c-modal-box--BorderSize) solid var(--pf-c-modal-box--BorderColor);
   box-shadow: var(--pf-c-modal-box--BoxShadow);
 
+  // At breakpoint--sm, set max width to variable value and ignore margins
+  &:not(.pf-m-lg) {
+    @media screen and (min-width: $pf-global--breakpoint--sm) {
+      max-width: var(--pf-c-modal-box--MaxWidth);
+    }
+  }
+
+  // At breakpoint--xl, set max width to variable value and ignore margins
   &.pf-m-lg {
-    max-width: var(--pf-c-modal-box--MaxWidth--lg);
+    @media screen and (min-width: $pf-global--breakpoint--xl) {
+      max-width: var(--pf-c-modal-box--lg--MaxWidth);
+    }
   }
 
+  // Header
   &__header {
     flex: 0 0 auto;
-    padding-top: var(--pf-c-modal-box__header--PaddingTop);
-    padding-right: var(--pf-c-modal-box__header--PaddingRight);
-    padding-bottom: var(--pf-c-modal-box__header--PaddingBottom);
-    padding-left: var(--pf-c-modal-box__header--PaddingLeft);
+    padding: var(--pf-c-modal-box__header--PaddingTop) var(--pf-c-modal-box__header--PaddingRight) var(--pf-c-modal-box__header--PaddingBottom) var(--pf-c-modal-box__header--PaddingLeft);
+
+    // Header title
+    &-title {
+      font-size: var(--pf-c-modal-box__header-title--FontSize);
+      font-weight: var(--pf-c-modal-box__header-title--FontWeight);
+      line-height: var(--pf-c-modal-box__header-title--LineHeight);
+
+      // If -webkit-line-clamp is supported
+      @supports (-webkit-line-clamp: var(--pf-c-modal-box--LineClamp)) {
+
+        // Up to max height $pf-c-modal-box--breakpoint-sm--Height: reset display and set single line overflow ellipsis
+        @media screen and (max-height: $pf-c-modal-box--breakpoint-sm--Height) {
+          overflow: hidden;
+          text-overflow: ellipsis;
+          white-space: nowrap;
+        }
+
+        // Between min height $pf-c-modal-box--breakpoint-sm--Height and max height $pf-c-modal-box--breakpoint-md--Height: set line clamp
+        @media screen and (min-height: $pf-c-modal-box--breakpoint-sm--Height) and (max-height: $pf-c-modal-box--breakpoint-md--Height) {
+          /* stylelint-disable */
+          display: -webkit-box;
+          overflow: hidden;
+          -webkit-line-clamp: var(--pf-c-modal-box--LineClamp);
+          -webkit-box-orient: vertical;
+          /* stylelint-enable */
+        }
+
+        // Up to max width $pf-global--breakpoint--sm: limit lines to 5
+        @media screen and (max-width: $pf-global--breakpoint--sm) and (min-height: $pf-c-modal-box--breakpoint-md--Height) {
+          /* stylelint-disable */
+          display: -webkit-box;
+          overflow: hidden;
+          -webkit-line-clamp: var(--pf-c-modal-box--MaxLineClamp);
```

Comment:
love this! thanks a lot.

would autoprefixer do the prefix  it for us?

if not lets do:
```scss
line-clamp: var(--pf-c-modal-box--MaxLineClamp);
-webkit-line-clamp: var(--pf-c-modal-box--MaxLineClamp);
```
to future proof this code

Reply:
Thanks! Autoprefixer appears not to take care of this. I'll include it.